staging/src/k8s.io/kubectl/pkg/kuberc/kuberc.go

```diff
@@ -0,0 +1,315 @@
+	}
+
+	for _, c := range kuberc.Spec.Overrides {
+		parsedCmds := strings.Split(c.Command, " ")
+		overrideCmd, _, err := rootCmd.Find(parsedCmds)
+		if err != nil {
+			// this may be referring to the alias command which is not initialized
+			// because the actual command is totally different.
+			return nil
+		}
+		if overrideCmd.Name() != cmd.Name() {
+			continue
+		}
+
+		if _, ok := p.aliases[cmd.Name()]; ok {
+			return fmt.Errorf("alias %s can not be overridden", cmd.Name())
+		}
+
+		// This function triggers merging the persistent flags in the parent commands.
+		_ = cmd.InheritedFlags()
+
+		for _, fl := range c.Flags {
+			// explicit flag usage has higher precedence than the kuberc default flag value.
+			// We should set the default flag values in kuberc, unless there is any in args.
+			if explicitFlagUse(fl.Name, args) {
+				continue
+			}
+			err = cmd.Flags().Set(fl.Name, fl.Default)
+			if err != nil {
+				return fmt.Errorf("could not apply value %s to flag %s in command %s err: %w", fl.Default, fl.Name, c.Command, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// ApplyAliases firstly appends all defined aliases in kuberc file to the root command.
+// Since there may be several alias definitions belonging to the same command, it extracts the
+// alias that is currently executed from args. After that it sets the flag definitions in alias as default values
+// of the command. Lastly, others parameters (e.g. resources, etc.) that are passed as arguments in kuberc
+// is appended into the os.Args.
+func (p *Preferences) ApplyAliases(rootCmd *cobra.Command, args []string, errOut io.Writer) ([]string, error) {
+	if len(args) <= 1 {
+		return args, nil
+	}
+
+	if !util.KubeRC.IsEnabled() {
+		return args, nil
+	}
+
+	_, _, err := rootCmd.Find(args[1:])
+	if err == nil {
+		// Command is found, no need to continue for aliasing
+		return args, nil
+	}
+
+	kubercPath := getExplicitKuberc(args)
+	kuberc, err := p.GetPreferencesFunc(kubercPath)
+	if err != nil {
+		return args, fmt.Errorf("kuberc error %w", err)
+	}
+
+	if kuberc == nil {
+		return args, nil
+	}
+
+	aliasArgsMap := make(map[string]struct {
+		args    []string
+		flags   []v1alpha1.PreferencesCommandOverrideFlag
+		command *cobra.Command
+	})
+
+	for _, alias := range kuberc.Spec.Aliases {
+		// do not allow shadowing built-ins
+		if _, _, err := rootCmd.Find([]string{alias.Name}); err == nil {
+			fmt.Fprintf(errOut, "Setting alias %q to a built-in command is not supported\n", alias.Name)
+			continue
+		}
+
+		if _, ok := aliasArgsMap[alias.Name]; ok {
+			fmt.Fprintf(errOut, "alias %s is already set, skipping...\n", alias.Name)
+			continue
+		}
+
+		commands := strings.Split(alias.Command, " ")
+		existingCmd, flags, err := rootCmd.Find(commands)
+		if err != nil {
+			fmt.Fprintf(errOut, "command %q not found to set alias %q: %v\n", alias.Command, alias.Name, flags)
+			continue
+		}
+
+		newCmd := *existingCmd
+		newCmd.Use = alias.Name
+		aliasCmd := &newCmd
```

Comment:
Recommendatin in https://github.com/spf13/cobra/issues/2147#issuecomment-2107566867 is applied

Reply:
cc: @marckhouzam